docs/reference/analysis/tokenfilters/synonym-graph-tokenfilter.asciidoc

```diff
@@ -0,0 +1,146 @@
+[[analysis-synonym-graph-tokenfilter]]
+=== Synonym Graph Token Filter
+
+experimental[]
+
+The `synonym_graph` token filter allows to easily handle synonyms, 
+including multi-word synonyms correctly during the analysis process. 
+
```

Comment:
Could you link here to the blog post the explains the complexities behind graph token streams?  http://blog.mikemccandless.com/2012/04/lucenes-tokenstreams-are-actually.html